cupy/manipulation/add_remove.py

```diff
@@ -13,7 +13,63 @@
 # TODO(okuta): Implement resize
 
 
-# TODO(okuta): Implement trim_zeros
+_first_non_zero_krnl = ReductionKernel(
+    'raw S data, T indices, T len, int64 mode',
+    'T y',
+    '''mode? data[indices] ? indices : len
+           : data[len - 1 - indices] ? len - 1 - indices : T(-1)''',
+    'mode? min(a,b): max(a, b) ',
+    'y = a',
+    'mode ? len : -1',
+    'first_non_zero'
+)
+_first_non_zero_cplx_krnl = ReductionKernel(
+    'raw S real, raw S imag, T indices, T len, int64 mode',
+    'T y',
+    '''mode? (real[indices] || imag[indices] ? indices : len)
+           : (real[len - 1 - indices] || imag[len - 1 - indices] ?
+              len - 1 - indices : T(-1))''',
+    'mode? min(a,b): max(a, b) ',
+    'y = a',
+    'mode ? len : -1',
+    'first_non_zero_cplx'
+)
+
+
+def trim_zeros(filt, trim='fb'):
+    """Trim the leading and/or trailing zeros from a 1-D array or sequence.
+
+    Returns the trimmed array
+
+    Args:
+        filt(1D array or sequence): Input array
+        trim(str, optional):
+            'fb' default option trims the array from both sides.
+            'f' option trim zeros from front.
+            'b' option trim zeros from front
+
+    Returns:
+        cupy.ndarray: trimmed input
+
+    .. seealso:: :func:`numpy.trim_zeros`
+    """
+    filt = cupy.asarray(filt)
```

Comment:
We don't want to support non-cupy-ndarray inputs to avoid unintended host-device synchronization.

Reply:
So list/sequence are not allowed  ? like trim_zeros([1,2,3])